### PR TITLE
refactored appointment list item component

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Summary
This PR refactors the appointment list item component to accept child components. This is the initial attempt at splitting a larger PR into a smaller PR.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#45823

## Testing done
Unit testing

## Screenshots
N/A

## What areas of the site does it impact?
VAOS

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx [hunk 1]
```diff
@@ -1,60 +1,29 @@
...
-        className="vaos-appts__card--clickable vads-u-margin-bottom--3"
         data-cy="appointment-list-item"
       >
-        <Card
-          appointment={appointment}
-          facility={facility}
-          link={link}
-          handleClick={() => handleClick({ history, link, idClickable })}
-          handleKeyDown={() => handleKeyDown({ history, link, idClickable })}
-        />
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:51]: Passed as a child to be more reusable.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx [hunk 2]
```diff
@@ -1,60 +1,29 @@
...
+}) {
   const idClickable = `id-${appointment.id.replace('.', '\\.')}`;
 
   return (
     <>
       <li
         id={idClickable}
+        className={className}
         data-request-id={appointment.id}
-        className="vaos-appts__card--clickable vads-u-margin-bottom--3"
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:42]: Passed in as a property to be more reusable.

File: src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx [hunk 3]
```diff
@@ -1,60 +1,29 @@
...
-        />
+        {children}
       </li>
     </>
   );
 }
 
 AppointmentListItem.propTypes = {
   appointment: PropTypes.object.isRequired,
-  facility: PropTypes.object,
```
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/components/AppointmentsPageV2/AppointmentListItem.jsx:59]: Removed since it has nothing to do with a list.

**Comment by simiadebowale** [comment]
AppointmentListGroup component is throwing an error. 
steps to reproduce:
- Click on the 'past' tab to view past appointments.
Screen Shot 2023-01-04 at 10 52 38 AM

**Comment by vbahinwillit** [comment]
Warning fixed. Thanks

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**